File: server/src/main/java/model/other/Spell.java
package model.other;

public class Spell implements ISpell {

    private String name;
    private String effect;
    private String type;
    private String cost;
    private String castTime;
    private ITalent talent;
    private int tier;

    public Spell() {
        this.name = "";
        this.effect = "";
        this.type = "";
        this.cost = "";
        this.castTime = "";
        this.talent = new Talent();
        this.tier = 1;
    }

    public int getTier() {
        return tier;
    }

    public void setTier(int tier) {
        this.tier = tier;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEffect() {
        return effect;
    }

    public void setEffect(String effect) {
        this.effect = effect;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getCost() {
        return cost;
    }

    public void setCost(String cost) {
        this.cost = cost;
    }

    public String getCastTime() {
        return castTime;
    }

    public void setCastTime(String castTime) {
        this.castTime = castTime;
    }

    public ITalent getTalent() {
        return talent;
    }

    public void setTalent(ITalent talent) {
        this.talent = talent;
    }
}

File: app/src/main/java/de/pnp/manager/app/model/Spell.kt
package de.pnp.manager.app.model

import de.pnp.manager.model.other.ISpell
import de.pnp.manager.model.other.ITalent

class Spell : ISpell {
    private var tier: Int = 0
    private var name: String? = null
    private var effect: String? = null
    private var type: String? = null
    private var cost: String? = null
    private var castTime: String? = null
    private var talent: ITalent? = null

    override fun getTier(): Int {
        return tier
    }

    override fun setTier(tier: Int) {
        this.tier = tier
    }

    override fun getName(): String {
        return this.name!!
    }

    override fun setName(name: String?) {
        this.name = name
    }

    override fun getEffect(): String {
        return this.effect!!
    }

    override fun setEffect(effect: String?) {
        this.effect = effect
    }

    override fun getType(): String {
        return this.type!!
    }

    override fun setType(type: String?) {
        this.type = type
    }

    override fun getCost(): String {
        return cost!!
    }

    override fun setCost(cost: String?) {
        this.cost = cost
    }

    override fun getCastTime(): String {
        return this.castTime!!
    }

    override fun setCastTime(castTime: String?) {
       this.castTime = castTime
    }

    override fun getTalent(): ITalent {
        return talent!!
    }

    override fun setTalent(talent: ITalent?) {
        this.talent = talent
    }
}